### PR TITLE
Improve VSCode DAP logpoint value summary

### PR DESCRIPTION
**Pull request by jeffreytan81**
Currently VSCode logpoint uses `SBValue::GetValue` to get the value for printing. This is not providing an intuitive result for std::string or char * -- it shows the pointer value instead of the string content.

This patch improves by prefers `SBValue::GetSummary()` before using `SBValue::GetValue()`.

### CODE REVIEW

File: lldb/tools/lldb-dap/BreakpointBase.cpp [hunk 1]
```diff
@@ -295,9 +295,11 @@ bool BreakpointBase::BreakpointHitCallback(
...
       if (value.GetError().Fail())
         value = frame.EvaluateExpression(expr);
-      const char *expr_val = value.GetValue();
-      if (expr_val)
-        output += expr_val;
+      llvm::StringRef summary_str = value.GetSummary();
+      if (!summary_str.empty())
+        output += summary_str.str();
+      else
+        output += value.GetValue();
```
Comment (clayborg) [inline lldb/tools/lldb-dap/BreakpointBase.cpp:302]: I think we should do what we do for variables: show the value + summary. If you have pointer, but it has a summary, we should still show the poiinter and tthe summary

Comment (jeffreytan81) [reply lldb/tools/lldb-dap/BreakpointBase.cpp:302]: Sure, I can do that.

File: lldb/test/API/tools/lldb-dap/breakpoint/TestDAP_logpoints.py [hunk 2]
```diff
@@ -48,7 +48,9 @@ def test_logmessage_basic(self):
         # 1. First at the loop line with logMessage
         # 2. Second guard breakpoint at a line after loop
         logMessage_prefix = "This is log message for { -- "
-        logMessage = logMessage_prefix + "{i + 3}"
+        message_addr_pattern = r"\b0x[0-9A-Fa-f]+\b"
+        message = '"Hello from main!"'
+        logMessage = logMessage_prefix + "{i + 3}, {message}"
```
Comment (bulbazord) [inline lldb/test/API/tools/lldb-dap/breakpoint/TestDAP_logpoints.py:53]: Does the string literal here need to be an f-string so `message` is properly interpolated?

File: lldb/tools/lldb-dap/BreakpointBase.cpp [hunk 3]
```diff
@@ -295,9 +296,7 @@ bool BreakpointBase::BreakpointHitCallback(
           frame.GetValueForVariablePath(expr, lldb::eDynamicDontRunTarget);
       if (value.GetError().Fail())
         value = frame.EvaluateExpression(expr);
-      const char *expr_val = value.GetValue();
-      if (expr_val)
-        output += expr_val;
+      output += ValeuToString(value);
```
Comment (bulbazord) [inline lldb/tools/lldb-dap/BreakpointBase.cpp:299]: Typo here -- `ValueToString`

File: lldb/tools/lldb-dap/JSONUtils.cpp [hunk 4]
```diff
@@ -210,8 +210,7 @@ static std::optional<std::string> TryCreateAutoSummary(lldb::SBValue value) {
   return TryCreateAutoSummaryForContainer(value);
 }
 
-void SetValueForKey(lldb::SBValue &v, llvm::json::Object &object,
-                    llvm::StringRef key) {
+std::string ValeuToString(lldb::SBValue v) {
```
Comment (bulbazord) [inline lldb/tools/lldb-dap/JSONUtils.cpp:213]: Same Typo

**Review by bulbazord — CHANGES_REQUESTED** [review]
Looks fine, but the method you added has a typo in its name.

**Review by bulbazord — APPROVED** [review]
Thanks!

File: lldb/test/API/tools/lldb-dap/breakpoint/TestDAP_logpoints.py (hunk 2 above)
Comment (jeffreytan81) [reply lldb/test/API/tools/lldb-dap/breakpoint/TestDAP_logpoints.py:51]: `{message}` is DAP's log message format not python f-string. Changed the test order and variable name to less confusing.

**Pull request merged**